mesh/mesh.hpp

```diff
@@ -106,6 +106,11 @@ class Mesh
    Array<Element *> boundary;
    Array<Element *> faces;
 
+   /// internal cache for element attributes
+   mutable Array<int> elem_attrs_cache;
+   /// internal cache for boundary element attributes
+   mutable Array<int> bdr_attrs_cache;
```

Comment:
I wonder if it makes sense to make these part of the `GeometricFactors` / `FaceGeometricFactors`.

Reply:
They can be requested with a new flag like `GeometricFactors::ATTRIBUTES`. This way things that one can get from an `ElementTransformation` (for 1 element) can be obtained from the `GeometricFactors` for all elements (or potentially sub-set / batch of elements in the future).

Reply:
My vote is to also keep them separate. They are not really geometric factors in the literal meaning of the word.

Reply:
As a user that uses both of those for very different things, I'd also prefer if they stayed separate. For example, I need to recompute the `GeometricFactors` once every newton iteration per time step in my code. However, I might need to change the attributes only once or twice during the entire execution of things.

Reply:
Okay, let's keep them as they are.